core/src/test/java/com/expedia/www/haystack/client/dispatchers/InMemoryDispatcher.java

```diff
@@ -0,0 +1,36 @@
+package com.expedia.www.haystack.client.dispatchers;
```

Comment:
Are you sure you want to write all these Dispatchers instead of just using Mockito? :-)

Reply:
This one is useful for a forthcoming feature and unit tests to validate it.  The NoOp ones could be mocked completely.